zerver/tests/test_messages.py

```diff
@@ -3336,12 +3336,19 @@ def test_update_invalid_flags(self) -> None:
                                    "flag": "mentioned"})
         self.assert_json_error(result, "Flag not editable: 'mentioned'")
 
-    def change_star(self, messages: List[int], add: bool=True, **kwargs: Any) -> HttpResponse:
-        return self.client_post("/json/messages/flags",
-                                {"messages": ujson.dumps(messages),
-                                 "op": "add" if add else "remove",
-                                 "flag": "starred"},
-                                **kwargs)
+    def change_star(self, messages: List[int], add: bool=True, bot_email: str=None,
+                    **kwargs: Any) -> HttpResponse:
+        if bot_email is None:
+            return self.client_post("/json/messages/flags",
+                                    {"messages": ujson.dumps(messages),
+                                     "op": "add" if add else "remove",
+                                     "flag": "starred"},
+                                    **kwargs)
+        return self.api_post(bot_email,
+                             "/json/messages/flags",
+                             {"messages": ujson.dumps(messages),
+                              "op": "add" if add else "remove",
+                              "flag": "starred"})
```

Comment:
This is sort of a weird test, in that bot users starring messages isn't an intentional feature.  While maybe it's fine to include the test here alongside the other `access_message` tests, we should include maybe write a straightforward test for this new behavior where the bot sends a message to a private stream it's not on, edits it, adds an emoji reaction, etc., while checking that a message sent just before by another user to that stream doesn't allow those same operations.

Reply:
I guess you've sorta written that in `test_delete_message_by_bot`; I would extend that test to try emoji reactions and editing as well (maybe renaming to `test_bot_access_to_send_messages_in_private_streams`.)